app/views/articles/index.html.erb

```diff
@@ -1,13 +1,19 @@
+<% if logged_in? %>
+    <h1>Welcome, <%= current_user.username %></h1>
+<%end%>
+
 <h1>List of Articles</h1>
 
 Our blog has <%= Article.public_count %> articles and counting!
 
 <ul>
   <% @articles.each do |article| %>
     <li>
-      <%= link_to article.title, article %>
+      <%= link_to article.title, article %> Created by:
+      <%= User.find_by(id: article.user_id).username %>
     </li>
   <% end %>
 </ul>
 
 <%= link_to "New Article", new_article_path %>
+<%= button_to "Logout", logout_path, method: :delete%>
```

Comment:
@idprietop if I am not logged in, I'm gonna see this logout button. Is that ok?

Reply:
If you have not logged in, you will not see the button because the before_action validates the session and if you have not logged in it redirects to the login.

Reply:
@idprietop that is the answer that I'm looking for! 👏🏻